src/mono/browser/runtime/loader/run.ts

```diff
@@ -125,6 +125,19 @@ export class HostBuilder implements DotnetHostBuilder {
         }
     }
 
+    // internal
+    withDumpThreadsOnExit(): DotnetHostBuilder {
```

Comment:
Since only for non-zero exit code, I would prefer `withDumpThreadsOnCrash` or `withDumpThreadsOnNonZeroExit`.
Also, as general rule, I think we should always add a boolean parameter to switch between on/off for these methods